modules/programs/zed-editor.nix

```diff
@@ -0,0 +1,97 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.zed-editor;
+  jsonFormat = pkgs.formats.json { };
+
+  configDir = if pkgs.stdenv.hostPlatform.isDarwin then
+    "Library/Application Support/zed" # TODO: see if this is actually the correct directory
+  else
+    "${config.xdg.configHome}/zed";
+  configFilePath = "${configDir}/settings.json";
+  keymapFilePath = "${configDir}/keymap.json";
+  mergedSettings = cfg.userSettings // {
+    auto_install_extensions = cfg.extensions;
```

Comment:
the `auto_install_extensions` configuration option doesn't work like this. it is a map of k/v pairs where the key is the extension name and the value is a boolean that determines if the extension is installed on startup.

so, if you want to keep this `extensions` option for the module as a QoL feature, since you're taking this merge approch already, you could instead use an expression to iterate over the provided extensions and provide the correct syntax. here's what i'm doing with a module i started work on before realising this PR existed :)

```nix
xdg.configFile."zed/settings.json" = mkIf (cfg.settings != { }) {
  source = (jsonFormat.generate "zed.json" (cfg.settings // {
    auto_install_extensions = lib.listToAttrs
      (map (ext: lib.nameValuePair ext true) cfg.extensions);
  }));
};
```

Reply:
thanks! do you want me to add you as a coauthor on that commit @cmacrae?

Reply:
very considerate of you, thanks! sure, that would be nice :)

Reply:
yw